keras/applications/imagenet_utils.py

```diff
@@ -58,6 +49,75 @@ def preprocess_input(x, data_format=None, mode='caffe'):
     return x
 
 
+def _bias_add_match_type(x, bias):
+    if K.dtype(x) != K.dtype(bias):
+        x = K.bias_add(x, K.cast(bias, K.dtype(x)))
```

Comment:
Explicitly pass `data_format` to `bias_add`. Otherwise it defaults to the global default and ignores the value passed to the parent.